waspc/data/Generator/templates/server/src/core/auth/prismaMiddleware.js

```diff
@@ -59,18 +60,25 @@ export const registerAuthMiddleware = (prismaClient) => {
   registerPasswordHashing(prismaClient)
 }
 
+const userValidations = [
+  { validates: PASSWORD_FIELD, message: 'password must be present', validator: password => !!password },
+  { validates: PASSWORD_FIELD, message: 'password must be at least 8 characters', validator: password => password.length >= 8 },
+  { validates: PASSWORD_FIELD, message: 'password must contain a number', validator: password => /\d/.test(password) },
+]
+
+{=# isUsernameAndPasswordAuthEnabled  =}
+userValidations.push({ validates: USERNAME_FIELD, message: 'username must be present', validator: username => !!username })
+{=/ isUsernameAndPasswordAuthEnabled  =}
+{=# isEmailAuthEnabled =}
+userValidations.push({ validates: USERNAME_FIELD, message: 'email must be present', validator: email => !!email })
+userValidations.push({ validates: USERNAME_FIELD, message: 'email must be valid', validator: email => isValidEmail(email) })
```

Comment:
Since our UI will probably say "E-mail" and the fact that this is called `username` will be an implementation detail, I'd rather keep the existing message.